cf-pubsub/src/main/java/org/eclipse/californium/pubsub/Topic.java

```diff
@@ -0,0 +1,202 @@
+/*******************************************************************************
+ * Copyright (c) 2018 RISE SICS and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ *
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ *
+ * Contributors:
+ *    Wajd Tohme
+ *    Ahmad Hussaein
+ *    Petr Kocián
+ *    Matias Carlander-Reuterfelt
+ *    Ismail Hilal
+ *    Tuna Gersil
+ *    Zainab Alsaadi
+ *
+ ******************************************************************************/
+package org.eclipse.californium.pubsub;
+
+import org.eclipse.californium.core.WebLink;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Set;
+
+
+/**
+ * The Topic class is a collection of functions that aims to make it easier to handle
+ * the results of a discover() function, by making sparse use of the Californium library
+ * through turning {@link WebLink} into Topic objects, that then can be stored in a
+ * sorted ArrayList. Topic objects have the extracted name, URI and Content Type of the
+ * Weblinks which you can then get individually.
+ */
+public class Topic {
+
+    String name;
+    String[] uri;
+    int ct;
+
+    /**
+     * Creates an instance of Topic given a WebLink from which it
+     * extracts the name, URI and Content Type
+     *
+     * @param wl WebLink
+     */
+    public Topic(WebLink wl) {
+        this.uri = wl.getURI().substring(1).split("(/)");
+        this.name = this.uri[this.uri.length - 1];
+        String corchete = wl.toString().substring(wl.toString().indexOf('[') + 1);
```

Comment:
I'm not sure, if this is reliable. It seems to read the first attribute value list, regardless of the attributes name. I would prefer a solution, which gets the attribute-value by name.

If that string-search approach is intended, then I would prefer to call `wl.toString()` only once and re-use it. 

Or